Type check seems broken in CI

It's showing a bunch of errors in files unrelated to a recent PR. https://github.com/quantumlib/Cirq/actions/runs/14840657094/job/41662582886?pr=7322

```
cirq-core/cirq/qis/entropy.py:78: error: Argument 1 to "_get_hamming_distance" has incompatible type "signedinteger[_8Bit]"; expected "ndarray[tuple[int, ...], dtype[signedinteger[_8Bit]]]"  [arg-type]
cirq-core/cirq/qis/entropy.py:78: error: Argument 2 to "_get_hamming_distance" has incompatible type "signedinteger[_8Bit]"; expected "ndarray[tuple[int, ...], dtype[signedinteger[_8Bit]]]"  [arg-type]
cirq-core/cirq/sim/simulation_utils.py:44: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], Any]", variable has type "ndarray[tuple[int], Any]")  [assignment]
cirq-core/cirq/sim/simulation_utils.py:45: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], Any]")  [assignment]
cirq-core/cirq/sim/simulation_utils.py:55: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], Any]", variable has type "ndarray[tuple[int], Any]")  [assignment]
cirq-core/cirq/sim/simulation_utils.py:59: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], Any]")  [assignment]
cirq-core/cirq/ops/controlled_operation.py:246: error: Invalid index type "tuple[object, ...]" for "ndarray[Any, Any]"; expected type "SupportsIndex | slice[Any, Any, Any] | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None | tuple[SupportsIndex | slice[Any, Any, Any] | EllipsisType | _SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]]]] | builtins.bool | int | _NestedSequence[builtins.bool | int] | None, ...]"  [index]
cirq-core/cirq/linalg/operator_spaces.py:83: error: Argument 1 to "__iadd__" of "ndarray" has incompatible type "ndarray[tuple[int, ...], dtype[complex[12](https://github.com/quantumlib/Cirq/actions/runs/14840657094/job/41662582886?pr=7322#step:5:13)8]] | ndarray[tuple[int, ...], dtype[number[Any, int | float | complex]]] | Any"; expected "_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]] | dtype[floating[Any]] | dtype[complexfloating[Any, Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]] | dtype[floating[Any]] | dtype[complexfloating[Any, Any]]]] | builtins.bool | int | float | complex | _NestedSequence[builtins.bool | int | float | complex]"  [arg-type]
cirq-core/cirq/experiments/qubit_characterizations.py:748: error: No overload variant of "__getitem__" of "Sequence" matches argument type "signedinteger[_32Bit | _64Bit]"  [call-overload]
cirq-core/cirq/experiments/qubit_characterizations.py:748: note: Possible overload variants:
cirq-core/cirq/experiments/qubit_characterizations.py:748: note:     def __getitem__(self, int, /) -> Sequence[SingleQubitCliffordGate]
cirq-core/cirq/experiments/qubit_characterizations.py:748: note:     def __getitem__(self, slice[Any, Any, Any], /) -> Sequence[Sequence[SingleQubitCliffordGate]]
cirq-core/cirq/experiments/qubit_characterizations.py:761: error: Argument 3 to "_two_qubit_clifford" has incompatible type "signedinteger[_32Bit | _64Bit]"; expected "int"  [arg-type]
cirq-core/cirq/work/observable_measurement_data.py:234: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[unsignedinteger[_8Bit]]]", variable has type "ndarray[tuple[int, int], dtype[unsignedinteger[_8Bit]]]")  [assignment]
cirq-core/cirq/work/observable_measurement_data.py:239: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[signedinteger[_64Bit]]]", variable has type "ndarray[tuple[int], dtype[signedinteger[_64Bit]]]")  [assignment]
cirq-core/cirq/work/observable_measurement_data.py:244: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], dtype[Any]]")  [assignment]
cirq-core/cirq/ops/linear_combinations.py:305: error: Argument 1 to "__iadd__" of "ndarray" has incompatible type "ndarray[tuple[int, ...], dtype[complex128]] | ndarray[tuple[int, ...], dtype[number[Any, int | float | complex]]] | Any"; expected "_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]] | dtype[floating[Any]] | dtype[complexfloating[Any, Any]]] | _NestedSequence[_SupportsArray[dtype[numpy.bool[builtins.bool]] | dtype[integer[Any]] | dtype[floating[Any]] | dtype[complexfloating[Any, Any]]]] | builtins.bool | int | float | complex | _NestedSequence[builtins.bool | int | float | complex]"  [arg-type]
cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py:291: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int, int], dtype[Any]]")  [assignment]
cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py:292: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], dtype[Any]]")  [assignment]
cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py:293: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], dtype[Any]]")  [assignment]
cirq-core/cirq/sim/clifford/stabilizer_state_ch_form.py:294: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], dtype[Any]]")  [assignment]
cirq-core/cirq/experiments/z_phase_calibration.py:245: error: Type argument "Sequence[Sequence[Axes]]" of "ndarray" must be a subtype of "tuple[int, ...]"  [type-var]
cirq-core/cirq/experiments/z_phase_calibration.py:249: error: Type argument "Sequence[Sequence[Axes]]" of "ndarray" must be a subtype of "tuple[int, ...]"  [type-var]
cirq-core/cirq/sim/simulation_utils_test.py:28: error: Argument 2 to "state_probabilities_by_indices" has incompatible type "list[signedinteger[_32Bit | _64Bit]]"; expected "Sequence[int]"  [arg-type]
cirq-core/cirq/sim/simulation_utils_test.py:29: error: Unsupported operand types for + ("list[signedinteger[_32Bit | _64Bit]]" and "list[int]")  [operator]
cirq-core/cirq/ops/arithmetic_operation_test.py:35: error: Incompatible types in assignment (expression has type "tuple[int, int]", variable has type "tuple[int, int, int, int]")  [assignment]
cirq-google/cirq_google/api/v1/programs.py:238: error: Incompatible types in assignment (expression has type "ndarray[tuple[int, ...], dtype[Any]]", variable has type "ndarray[tuple[int], dtype[Any]]")  [assignment]
Found 24 errors in 12 files (checked 1[14](https://github.com/quantumlib/Cirq/actions/runs/14840657094/job/41662582886?pr=7322#step:5:15)8 source files)

Error: Process completed with exit code 1.
```

--- a/dev_tools/requirements/mypy.env.txt
+++ b/dev_tools/requirements/mypy.env.txt
@@ -4,3 +4,6 @@
 
 -r deps/cirq-all.txt
 -r deps/mypy.txt
+
+# TODO: #7323 - remove after fixing type annotations for NumPy-2
+numpy<2.0

